service/src/test/java/com/commercetools/pspadapter/tenant/TenantFactoryTest.java

```diff
@@ -0,0 +1,75 @@
+package com.commercetools.pspadapter.tenant;
+
+import com.commercetools.pspadapter.payone.config.PayoneConfig;
+import com.commercetools.pspadapter.payone.domain.ctp.CustomTypeBuilder;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TenantFactoryTest {
+
+    @Mock
+    private PayoneConfig payoneConfig;
+
+    @Mock
+    private TenantConfig tenantConfig;
+
+    private TenantFactory factory;
+
+    @Before
+    public void setUp() throws Exception {
+        when(payoneConfig.getApiUrl()).thenReturn("http://test.api.url");
+        when(tenantConfig.getPayoneConfig()).thenReturn(payoneConfig);
+        when(tenantConfig.getName()).thenReturn("testTenantName");
+
+        factory = new TenantFactory("testPayoneInterfaceName", tenantConfig);
+    }
+
+    @Test
+    public void getsTenantName() throws Exception {
+        assertThat(factory.getTenantName()).isEqualTo("testTenantName");
+    }
+
+    @Test
+    public void getsPayoneInterfaceName() throws Exception {
+        assertThat(factory.getPayoneInterfaceName()).isEqualTo("testPayoneInterfaceName");
+    }
+
+    @Test
+    public void getsPaymentHandlerUrl() throws Exception {
+        assertThat(factory.getPaymentHandlerUrl()).contains("/testTenantName/");
```

Comment:
but as I see here `getPaymentHandlerUrl()` concatenates the url prefix, which seems in this case `/testTenantName`, to hardcoded string `"commercetools/handle/payments/:id"`. So both the hardcoded 'base URL' and the concatenation are coming from this unit (`TenantFactory`).

What if the some dev changes this hardcoded string by mistake, this unit test will still succeed when it shouldn't.